### PR TITLE
Moving boxes example thorn

### PR DESCRIPTION
**Pull request by lucass-carneiro**
This PR provides a new thorn called `MovingBoxesToy` that sets up two refinement boxes and makes them move around in a circle. It is intended as a guide for people wanting to implement more complex moving boxes.

In order to make this thorn possible, I had to change the parameters in `BoxInBox` to `RESTRICTED`

### CODE REVIEW

**Comment by eschnett** [comment]
Thanks! This thorn is a good idea.

I don't think that `BoxInBox`'s parameters should be "restricted", i.e. should be visible outside the thorn. If `MovingBoxToy` wants to move the box around then it should decide where the origin of the circle is. Using the position from `BoxInBox` is probably just confusing. Another reason is that these parameters are only used to initialize grid scalars, and looking at these parameters later on will give wrong information.

**Comment by lucass-carneiro** [comment]
Thanks for the feedback! I have removed the `RESTRICTED` from `BoxInBox` and changed the toy. The boxes now move without requiring parameters from `BoxInBox`

**Comment by eschnett** [comment]
You are currently not running the test cases in CI since you are not updating the thorn list in the `scripts` directory.

**Comment by lucass-carneiro** [comment]
I have updated the thorn list inside `scripts`

**Comment by eschnett** [comment]
There are no test cases...

File: MovingBoxToy/test/circle/performance.yaml
Comment (eschnett) [inline MovingBoxToy/test/circle/performance.yaml:1]: Don't add this file, its contents are not stable.

File: MovingBoxToy/test/circle.par [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+BoxInBox::radius_z_2[1] = 0.25
+
+CarpetX::boundary_x = "dirichlet"
+CarpetX::boundary_y = "dirichlet"
+CarpetX::boundary_z = "dirichlet"
+CarpetX::boundary_upper_x = "dirichlet"
+CarpetX::boundary_upper_y = "dirichlet"
+CarpetX::boundary_upper_z = "dirichlet"
+
+Cactus::cctk_itlast = 20
```
Comment (eschnett) [inline MovingBoxToy/test/circle.par:44]: 20 iterations is a lot. Would 5 iterations suffice? That would make this test case 4 times smaller.

**Review by eschnett — CHANGES_REQUESTED** [review]

**Comment by lucass-carneiro** [comment]
I've removed the `yml` file and reduce the size of the test

**Review by eschnett — APPROVED** [review]

**Pull request merged**